File: server/router/normal/gost_client_p2p.go
package normal

import (
	"github.com/gin-gonic/gin"
	"server/controller/normal/gost_client_p2p"
	"server/global"
	"server/router/middleware"
)

func InitGostClientP2P(group *gin.RouterGroup) {
	g := group.Group("gost/client/p2p", middleware.Auth(global.Jwt))
	g.POST("create", gost_client_p2p.Create)
	g.POST("renew", gost_client_p2p.Renew)
	g.POST("migrate", gost_client_p2p.Migrate)
	g.POST("update", gost_client_p2p.Update)
	g.POST("enable", gost_client_p2p.Enable)
	g.POST("delete", gost_client_p2p.Delete)
	g.POST("page", gost_client_p2p.Page)
}

File: web/src/api/normal/gost_client_p2p.js
import request from "../../request/index.js";

const baseUrl = '/v1/normal/gost/client/p2p'

export const apiNormalGostClientP2PList = () => {
    return request.request({
        url: `${baseUrl}/list`,
        method: 'POST',
    })
}

export const apiNormalGostClientP2PPage = (data) => {
    return request.request({
        url: `${baseUrl}/page`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientP2PCreate = (data) => {
    return request.request({
        url: `${baseUrl}/create`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientP2PRenew = (data) => {
    return request.request({
        url: `${baseUrl}/renew`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientP2PMigrate = (data) => {
    return request.request({
        url: `${baseUrl}/migrate`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientP2PEnable = (data) => {
    return request.request({
        url: `${baseUrl}/enable`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientP2PUpdate = (data) => {
    return request.request({
        url: `${baseUrl}/update`,
        method: 'POST',
        data
    })
}


export const apiNormalGostClientP2PDelete = (data) => {
    return request.request({
        url: `${baseUrl}/delete`,
        method: 'POST',
        data
    })
}